### ISSUE TITLE
Allow using a nickname instead of script name for characters in dialogs

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags#1274]
A character may have a name with spaces or other characters that are invalid or not convenient for the character script name.

Ideally when writing dialogs I would like to write the character name as is writing in the game script (as in play not in code). This also makes easier to revise the game dialogs since it's similar to the form a writer would have provided the script.

This would give a good logic to follow if the existing logic of today is removed with #1273 .

**Comment by AlanDrake** [comment]
Non-script names aren't necessarily unique, so using the script name is the pragmatically correct choice.

**Comment by ericoporto** [comment]
A `cSarahOld` I could override as `OLD SARAH CONNOR` so the dialog script looks like:
```
OLD SARAH CONNOR: The Terminator changed everything... 
```

**Comment by ivan-mogilko** [comment]
So, what needs to be specified here:
1) Which property to use? Is it Character's RealName, or a new property called something like DialogName, as suggested in [this comment](https://github.com/adventuregamestudio/ags/issues/1273#issuecomment-839637649) to #1273.
2) Should that property be saved into game data, is there any need for that; or is it enough to keep at project level, and replace with script name at the dialog->script conversion stage.
